Fix namespaces not respecting accessible namespace config

Nokel81 opened the pull request:
Signed-off-by: Sebastian Malton <sebastian@malton.name>

minimal version of #7277 

### Description of change

- Due to #6657 `NamespaceStore` was changed to no longer include fake loading and fake instances of the namespaces retrieved from the `refreshAccessbility` logic within `Cluster`. Thus `NamespaceStore.items` is not `[]` if the user does not have `list` permissions for `namespaces` (just like all other resources).
- `NamespaceStore`’s dependencies field has a context field. This is set to `clusterFrameContextForClusterScopedResources` (since it is a cluster scoped resource)
- As such `this.dependencies.context.allNamespaces` is always `[]`, so reverting the implementation of `allowedNamespaces` changed nothing

It is preferable to not have a self-referential updating stateful data source as it has made `NamespaceStore` extremely  hard to reason about.

Nokel81 committed:
Log state after refresh accessibility

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Nokel81 committed:
Fix namespaces not respecting accessible namespace config

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Nokel81 committed:
Fix warning hover text formatting

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Nokel81 committed:
Fix tests

Signed-off-by: Sebastian Malton <sebastian@malton.name>

jakolehm commented on packages/core/src/renderer/components/+workloads-pods/load-pods-from-all-namespaces.injectable.ts:
```diff
@@ -4,20 +4,20 @@
     return () => {
       podStore.loadAll({
-        namespaces: [...namespaceStore.getItems().map(ns => ns.getName())],
+        namespaces: context.allNamespaces,
```
Why do we need to pass `namespaces` here, shouldn't it default to all namespaces if none given?

Nokel81 committed:
Fix NamespaceStore.allowedNamespaces

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Nokel81 committed:
Remove unnecessary '?'

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Nokel81 committed:
Add deprecation messages

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Nokel81 committed:
Move selected_namespaces storage to injectable

- And its initialization

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Nokel81 committed:
Implement contextNamespaces logic in forNamespacedResources

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Nokel81 committed:
Update snapshot

Signed-off-by: Sebastian Malton <sebastian@malton.name>

jakolehm commented on packages/core/src/common/cluster/cluster.ts:
```diff
@@ -498,6 +498,7 @@ export class Cluster implements ClusterModel {
 
     this.allowedResources.replace(await this.getAllowedResources(requestNamespaceListPermissions));
     this.ready = this.knownResources.length > 0;
+    this.dependencies.logger.info(`[CLUSTER]: refreshed accessibility data`, this.getState());
```
Maybe `debug`?

Nokel81 committed:
Fix formatting

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Nokel81 committed:
Fix formatting

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Nokel81 committed:
Fix test

- This was a side effect of the previous bug fixes where
  the clusterContext.hasAllSelected was previously erroneously 'false'

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Nokel81 committed:
Change log level

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Nokel81 committed:
Fix another test suite

Signed-off-by: Sebastian Malton <sebastian@malton.name>
